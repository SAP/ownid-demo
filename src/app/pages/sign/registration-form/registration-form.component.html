<form [formGroup]="form" (ngSubmit)="register()" >
<p>
  <span class="errors" *ngIf="errors">{{errors}}</span>
</p>
<p>
  <mat-form-field class="form-field">
    <mat-label>First Name</mat-label>
    <input matInput placeholder="Joe" formControlName="firstName" />
  </mat-form-field>
</p>
<p>
  <mat-form-field class="form-field">
    <mat-label>Last Name</mat-label>
    <input matInput placeholder="Speen" formControlName="lastName" />
  </mat-form-field>
</p>
<p>
  <mat-form-field class="form-field">
    <mat-label>Enter your email</mat-label>
    <input
      matInput
      placeholder="pat@example.com"
      formControlName="email"
      required
    />
    <mat-error *ngIf="form.controls.email.hasError('required')"
      >You must enter a value</mat-error
    >
    <mat-error *ngIf="form.controls.email.hasError('email')">Not a valid email</mat-error>
  </mat-form-field>
</p>
<p>
  <mat-form-field class="form-field">
    <mat-label>Enter your password</mat-label>
    <input matInput [type]="hide ? 'password' : 'text'" formControlName="password" required />
    <button
      mat-icon-button
      matSuffix
      (click)="hide = !hide"
      [attr.aria-label]="'Hide password'"
      [attr.aria-pressed]="hide"
    >
      <mat-icon>{{ hide ? 'visibility_off' : 'visibility' }}</mat-icon>
    </button>
  </mat-form-field>
</p>
<p>
  <mat-form-field class="form-field">
    <mat-label>Confirm password</mat-label>
    <input matInput [type]="hideConfirm ? 'password' : 'text'" required />
    <button
      mat-icon-button
      matSuffix
      (click)="hideConfirm = !hideConfirm"
      [attr.aria-label]="'Hide password'"
      [attr.aria-pressed]="hideConfirm"
    >
      <mat-icon>{{ hideConfirm ? 'visibility_off' : 'visibility' }}</mat-icon>
    </button>
  </mat-form-field>
</p>
<p>
  <button type="submit" mat-flat-button color="primary" class="form-field">
    Create Account
  </button>
</p>
</form>
