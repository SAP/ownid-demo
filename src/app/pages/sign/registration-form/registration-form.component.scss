:host {
  display: block;
  margin-top: 36px;
}

.form-field {
  width: 100%;
}

.forgot-link {
  color: #777777;
}

.errors {
  color: red;
  font-size: 14px;
  line-height: 18px;
  margin-bottom: 16px;
}
