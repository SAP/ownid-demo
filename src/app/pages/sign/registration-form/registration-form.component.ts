import { Component, ChangeDetectionStrategy } from "@angular/core";
import { FormControl, Validators } from "@angular/forms";

@Component({
  selector: "registration-form",
  templateUrl: "./registration-form.component.html",
  styleUrls: ["./registration-form.component.scss"],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class RegistrationFormComponent {
  email = new FormControl("", [Validators.required, Validators.email]);

  firstName = new FormControl("", [Validators.required]);

  lastName = new FormControl("", [Validators.required]);

  hide = true;

  hideConfirm = true;
}
