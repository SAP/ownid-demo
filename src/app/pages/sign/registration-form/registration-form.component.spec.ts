import { async, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { GigyaService } from '@services/gigya.service';
import { FormBuilder } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { Router } from '@angular/router';
import { RegistrationFormComponent } from './registration-form.component';

describe('RegistrationFormComponent', () => {
  let gigyaService: GigyaService;
  let router: Router;

  beforeEach(() => {
    gigyaService = {} as GigyaService;
    gigyaService.register = jest.fn();
    router = {} as Router;
    router.navigateByUrl = jest.fn();
  });

  describe('Snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        providers: [
          FormBuilder,
          {
            provide: GigyaService,
            useValue: gigyaService,
          },
        ],
        declarations: [RegistrationFormComponent],
        schemas: [NO_ERRORS_SCHEMA],
      }).compileComponents();
    }));

    test('check snapshot', () => {
      const fixture = TestBed.createComponent(RegistrationFormComponent);

      expect(fixture).toMatchSnapshot();
    });
  });

  describe('register', () => {
    it('should call gigyaService.register and redirect to /account if success', () => {
      const sut = new RegistrationFormComponent(new FormBuilder(), gigyaService, router);
      sut.form.patchValue({ email: 'test@mail.com', password: '1234567' });
      return new Promise((resolve) => {
        gigyaService.register = jest.fn().mockImplementation((data, c) => {
          expect(data).not.toBeNull();
          c({ status: 'OK' });
          expect(gigyaService.register).toBeCalled();
          expect(router.navigateByUrl).toBeCalledWith('/account');
          resolve();
        });
        sut.register();
      });
    });

    it('should call gigyaService.register and redirect set errors if failed', () => {
      return new Promise((resolve) => {
        const sut = new RegistrationFormComponent(new FormBuilder(), gigyaService, router);
        sut.form.patchValue({ email: 'test@mail.com', password: '1234567' });
        const errorDetails = 'fakeDetails';
        gigyaService.register = jest.fn().mockImplementation((data, c) => {
          expect(data).not.toBeNull();
          c({ status: 'FAIL', errorDetails });
          expect(gigyaService.register).toBeCalled();
          expect(sut.errors).toBe(errorDetails);
          resolve();
        });
        sut.register();
      });
    });

    it('should do nothing if form validation fails', () => {
      const sut = new RegistrationFormComponent(new FormBuilder(), gigyaService, router);
      sut.form.patchValue({ email: 'testil.com', password: '' });
      sut.register();
      expect(gigyaService.register).not.toBeCalled();
      expect(sut.errors).toBe(null);
    });
  });

  describe('onSuccessOwnIdResgister', () => {
    it('should do nothing if form validation fails', () => {
      const sut = new RegistrationFormComponent(new FormBuilder(), gigyaService, router);
      const resp = { data: {} };
      sut.onSuccessOwnIdResgister(resp);
      expect(sut.showForm).toBe(true);
      expect(sut.registerResponse).toBe(resp);
    });
  });
});
