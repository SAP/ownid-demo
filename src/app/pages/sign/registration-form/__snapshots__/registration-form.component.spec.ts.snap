// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationFormComponent Snapshot check snapshot 1`] = `
<registration-form
  errors="null"
  form={[Function FormGroup]}
  gigyaService={[Function Object]}
  hide={[Function Boolean]}
  hideConfirm={[Function Boolean]}
  router={[Function Router]}
>
  <form>
    <p>
      
    </p>
    <p>
      <mat-form-field
        class="form-field"
      >
        <mat-label>
          First Name
        </mat-label>
        <input
          formcontrolname="firstName"
          matinput=""
          placeholder="Joe"
        />
      </mat-form-field>
    </p>
    <p>
      <mat-form-field
        class="form-field"
      >
        <mat-label>
          Last Name
        </mat-label>
        <input
          formcontrolname="lastName"
          matinput=""
          placeholder="Speen"
        />
      </mat-form-field>
    </p>
    <p>
      <mat-form-field
        class="form-field"
      >
        <mat-label>
          Enter your email
        </mat-label>
        <input
          formcontrolname="email"
          matinput=""
          placeholder="pat@example.com"
          required=""
        />
        
        
      </mat-form-field>
    </p>
    <p>
      <mat-form-field
        class="form-field"
      >
        <mat-label>
          Enter your password
        </mat-label>
        <input
          formcontrolname="password"
          matinput=""
          required=""
        />
        <button
          mat-icon-button=""
          matsuffix=""
        >
          <mat-icon>
            
          </mat-icon>
        </button>
      </mat-form-field>
    </p>
    <p>
      <mat-form-field
        class="form-field"
      >
        <mat-label>
          Confirm password
        </mat-label>
        <input
          matinput=""
          required=""
        />
        <button
          mat-icon-button=""
          matsuffix=""
        >
          <mat-icon>
            
          </mat-icon>
        </button>
      </mat-form-field>
    </p>
    <p>
      <button
        class="form-field"
        color="primary"
        mat-flat-button=""
        type="submit"
      >
         Create Account 
      </button>
    </p>
  </form>
</registration-form>
`;
